### ISSUE TITLE
Reusable Packages

### ISSUE THREAD

**Issue by danielBingham** [danielBingham/peerreview#151]
## Story Description 
*What are we trying to accomplish in plain english?*

In order to do #147, we're going to need the ability to break our backend into packages that can be used by both the `web-application` and by the `workers`.  

## Acceptance Criteria
*At what point is this story considered "done"?*

- We have a `backend` package that includes the models and services.
- We have a way to include the `backend` package in both the `web-application` and in the `workers`.

## Dependencies
*What stories does this one depend on?  What do we need to do first before we can call this one done?*

## Task List

- [ ] Explore using [lerna](https://lerna.js.org/docs/getting-started) for package building and publishing.
- [ ] Explore using [Github Packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry) as our package repository.
- [ ] Update the [CI](https://github.com/danielBingham/peerreview/actions/workflows/ci.yaml) workflow to be Main - CI.
- [ ] Update the Main - CI workflow to account for package building and publishing.
- [ ] Update the `web-application/Dockerfile` to pull the backend package from our chosen package repo.
- [ ] Create a new PR - CI workflow to handle CI on PRs and publish packages based on branches.

**Comment by danielBingham** [comment]
We can create packages pretty easily: [Creating node.js modules](https://docs.npmjs.com/creating-node-js-modules).

One question is whether we can simply keep this as a local package and build the containers locally or whether we need to push it up to the Github Package repository for it to be available to Github Actions for use in the CI build.  This seems unclear.

**Comment by danielBingham** [comment]
- [ ] Explore using [lerna](https://lerna.js.org/docs/getting-started) for package building and publishing.
- [ ] Explore using [Github Packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry) as our package repository.
- [ ] Update the [CI](https://github.com/danielBingham/peerreview/actions/workflows/ci.yaml) workflow to be Main - CI.
- [ ] Update the Main - CI workflow to account for package building and publishing.
- [ ] Update the `web-application/Dockerfile` to pull the backend package from our chosen package repo.
- [ ] Create a new PR - CI workflow to handle CI on PRs and publish packages based on branches.
